src/interfaces/utils-interface.ts

```diff
@@ -0,0 +1,10 @@
+// Interface to define the JWT
+export interface JwtIF {
+  roles: Array<string>
+}
+
+export interface AuthIF {
+  data: {
```

Comment:
Curious if this IF allows a variable of this type to be null?

Reply:
I believe without making it optional or giving it `| null`, it would expect a value.

Reply:
So you'd have to initialize (empty) such a variable to = { data: { roles: [ ] } } ?

I'm still learning how to use interfaces -- I think they're like structs -- but I'm thinking it would be nice to be able to initial the variable as null.  It may be possible as-is. Thoughts?

Reply:
Great question for sure! Interfaces are just describing what we, as developers, expect to see within the object. It certainly doesn't always paint the whole picture, as there can be potentially be MUCH more than what we describe in the interface. 

So in this instance specifically, we don't need to initialize the object with empty values, just define what we expect, because we are assigning this type to a response object from the API. At this point, the only thing we expect from this response is { data: { roles: Array<string> } }.  We could make data optional, but i believe the way we are using it, it shouldn't be, as the error handling will throw Error if the data property isn't present in the response.

Reply:
Did any of that explanation help :) ?